src/main/java/com/microsoft/jenkins/kubernetes/V1ResourceManager.java

```diff
@@ -0,0 +1,599 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.jenkins.kubernetes;
+
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.apis.AppsV1Api;
+import io.kubernetes.client.apis.BatchV1Api;
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1ConfigMap;
+import io.kubernetes.client.models.V1DaemonSet;
+import io.kubernetes.client.models.V1Deployment;
+import io.kubernetes.client.models.V1Job;
+import io.kubernetes.client.models.V1Namespace;
+import io.kubernetes.client.models.V1Pod;
+import io.kubernetes.client.models.V1ReplicaSet;
+import io.kubernetes.client.models.V1ReplicationController;
+import io.kubernetes.client.models.V1Secret;
+import io.kubernetes.client.models.V1Service;
+import io.kubernetes.client.models.V1ServicePort;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class V1ResourceManager extends ResourceManager {
+    private CoreV1Api coreV1ApiInstance;
+    private AppsV1Api appsV1ApiInstance;
+    private BatchV1Api batchV1ApiInstance;
+    private String pretty = DEFAULT_PRETTY;
+    private V1ResourceUpdateMonitor resourceUpdateMonitor = V1ResourceUpdateMonitor.NOOP;
+
+    public V1ResourceManager() {
+        coreV1ApiInstance = new CoreV1Api();
```

Comment:
Move these APIClients to a singleton pattern?

Reply:
Agree, changed.